lms/djangoapps/grades/api/tests/test_views.py

```diff
@@ -0,0 +1,259 @@
+        self.assertEqual(resp.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_self_get_grade(self):
+        """
+        Test that a student can request her own grade
+        """
+        resp = self.client.get(self.get_url(self.student.username))
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_nonexistent_user(self):
+        """
+        Test that a nonexistent username does not return anything
+        """
+        resp = self.client.get(self.get_url('IDoNotExist'))
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertIn('error_code', resp.data)  # pylint: disable=no-member
+        self.assertEqual(resp.data['error_code'], 'user_mismatch')  # pylint: disable=no-member
+
+    def test_other_get_grade(self):
+        """
+        Test that a student cannot get the grade of another student
+        """
+        self.client.logout()
+        self.client.login(username=self.other_student.username, password=self.password)
+        resp = self.client.get(self.get_url(self.student.username))
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertIn('error_code', resp.data)  # pylint: disable=no-member
+        self.assertEqual(resp.data['error_code'], 'user_mismatch')  # pylint: disable=no-member
+
+    def test_self_get_grade_no_enrolled(self):
+        """
+        Test that a student cannot request her own grade if not enrolled
+        """
+        # a user not enrolled in the course cannot request her grade
+        self.client.logout()
+        self.client.login(username=self.other_user.username, password=self.password)
+        resp = self.client.get(self.get_url(self.other_user.username))
+        self.assertEqual(resp.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertIn('error_code', resp.data)  # pylint: disable=no-member
+        self.assertEqual(
+            resp.data['error_code'],  # pylint: disable=no-member
+            'user_or_course_does_not_exist'
+        )
+
+    def test_wrong_course_key(self):
+        """
+        Test if the course key does not exist
+        """
+        # from nose.tools import set_trace; set_trace()
+        def mock_from_string(*args, **kwargs):  # pylint: disable=unused-argument
+            """Mocked function to always raise an exception"""
+            raise InvalidKeyError('foo', 'bar')
+        with patch('opaque_keys.edx.keys.CourseKey.from_string', side_effect=mock_from_string):
+            resp = self.client.get(self.get_url(self.student.username))
+        assert resp.status_code == status.HTTP_404_NOT_FOUND
+        self.assertIn('error_code', resp.data)  # pylint: disable=no-member
+        self.assertEqual(
+            resp.data['error_code'],  # pylint: disable=no-member
+            'invalid_course_key'
+        )
+
+    def test_couse_does_not_exist(self):
+        """
+        Test if the course key is valid but the course does not exist
```

Comment:
For several of the tests around here, the docstring is explaining the situation we're testing (the input), but not the expected result (output). Can you update them to include that information?

For example, this method could say `Test that requesting a valid, nonexistent course key returns an error as expected`